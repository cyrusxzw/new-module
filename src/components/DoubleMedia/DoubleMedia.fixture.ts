const DoubleMediaFixture = {
  imageOne: {
    type: 'image',
    altText:
      'A tactile approach infuses our work, from our product development to our retail service and visual merchandising.',
    caption:
      'Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    sizes: {
      large: './assets/images/DoubleImage/image-one-large.jpg',
      medium: './assets/images/DoubleImage/image-one-medium.jpg',
      small: './assets/images/DoubleImage/image-one-small.jpg',
    },
  },
  imageTwo: {
    type: 'image',
    altText:
      'Image of: Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    caption:
      'Our passion for creativity in form and function, extends to the setting of a meal among staff.',
    sizes: {
      large: './assets/images/DoubleImage/image-two-large.jpg',
      medium: './assets/images/DoubleImage/image-two-medium.jpg',
      small: './assets/images/DoubleImage/image-two-small.jpg',
    },
  },
  link: {
    text: 'Cleanse',
    url: '/au/c/skin/cleanse',
    hasTargetInNewWindow: null,
    type: 'Relative',
  },
  caption:
    'Our passion for creativity in form and function, extends to the setting of a meal among staff.',
  video: {
    type: 'video',
    poster: {
      copy: {
        altText: 'How to Use Video for Tea Tree Leaf Facial Exfoliant',
      },
      sizes: {
        large: './assets/images/MediaWithContent/video-poster.png',
        medium: './assets/images/MediaWithContent/video-poster.png',
        small: './assets/images/MediaWithContent/video-poster.png',
      },
    },
    fallbackImage: {
      copy: {
        altText: 'How to Use Video for Tea Tree Leaf Facial Exfoliant',
      },
      sizes: {
        large: './assets/images/MediaWithContent/video-poster.png',
        medium: './assets/images/MediaWithContent/video-poster.png',
        small: './assets/images/MediaWithContent/video-poster.png',
      },
    },
    sizes: {
      large:
        'https://player.vimeo.com/external/225136214.hd.mp4?s=a01cd8b47f35767394e48dc01af10510ef8ff4dd&profile_id=174',
      medium:
        'https://player.vimeo.com/external/225136214.hd.mp4?s=a01cd8b47f35767394e48dc01af10510ef8ff4dd&profile_id=174',
      small:
        'https://player.vimeo.com/external/225136214.hd.mp4?s=a01cd8b47f35767394e48dc01af10510ef8ff4dd&profile_id=174',
    },
  },
  title: 'Aesop favourites',
};

export { DoubleMediaFixture };
