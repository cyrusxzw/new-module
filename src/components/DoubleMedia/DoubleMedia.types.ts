import type { Themes, ComponentWithoutChildren } from '~/types';
import type { MediaBlockProps } from './components/MediaBlock/MediaBlock.types';

type DoubleMediaProps = {
  className?: string;
  mediaOne?: MediaBlockProps;
  mediaTwo?: MediaBlockProps;
  theme?: Themes;
};

type DoubleMediaType = ComponentWithoutChildren<DoubleMediaProps>;

export type { DoubleMediaType, DoubleMediaProps };
