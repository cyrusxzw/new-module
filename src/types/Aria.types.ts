type Aria = {
  expanded?: boolean;
  haspopup?: boolean | 'dialog' | 'menu' | 'listbox' | 'tree' | 'grid';
  hidden?: boolean;
  label?: string;
};

export type { Aria };
