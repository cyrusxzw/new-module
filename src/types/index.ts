export type { Product } from './Product.types';
export type { Themes } from './Themes.types';
export type { Variant } from './Variant.types';
export type { ViewportNames, BreakpointNames } from './Viewport.types';
export type { OnAddToCartClick } from '~/contexts/AddToCartContext/AddToCartContext.types';
