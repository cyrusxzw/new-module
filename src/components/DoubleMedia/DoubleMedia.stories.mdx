import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { DoubleMediaFixture } from './DoubleMedia.fixture';
import { DoubleMedia } from './DoubleMedia';

<Meta title="Components/DoubleMedia" component={DoubleMedia} />

# DoubleMedia

<Canvas>
  <Story name="Base component">
    <DoubleMedia
      mediaOne={{
        heading: knobs.text('Title 1', 'Title 1', 'Content'),
        link: knobs.object('Link 1', DoubleMediaFixture.link, 'Links'),
        type: DoubleMediaFixture.imageOne.type,
        altText: DoubleMediaFixture.imageOne.altText,
        caption: DoubleMediaFixture.imageOne.caption,
        sizes: DoubleMediaFixture.imageOne.sizes,
      }}
      mediaTwo={{
        heading: knobs.text('Title 2', 'Title 2', 'Content'),
        link: knobs.object('Link 2', DoubleMediaFixture.link, 'Links'),
        type: DoubleMediaFixture.imageTwo.type,
        altText: DoubleMediaFixture.imageTwo.altText,
        caption: DoubleMediaFixture.imageTwo.caption,
        sizes: DoubleMediaFixture.imageTwo.sizes,
      }}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Canvas>

## Variations

### Video in slots

<Canvas>
  <Story name="Contains videos">
    <DoubleMedia
      mediaOne={{
        heading: knobs.text('Title 1', 'Title 1', 'Content'),
        link: knobs.object('Link 1', DoubleMediaFixture.link, 'Links'),
        type: DoubleMediaFixture.video.type,
        caption: DoubleMediaFixture.caption,
        poster: DoubleMediaFixture.video.poster,
        fallbackImage: DoubleMediaFixture.video.fallbackImage,
        sizes: DoubleMediaFixture.video.sizes,
      }}
      mediaTwo={{
        heading: knobs.text('Title 2', 'Title 2', 'Content'),
        link: knobs.object('Link 2', DoubleMediaFixture.link, 'Links'),
        type: DoubleMediaFixture.video.type,
        caption: DoubleMediaFixture.caption,
        poster: DoubleMediaFixture.video.poster,
        fallbackImage: DoubleMediaFixture.video.fallbackImage,
        sizes: DoubleMediaFixture.video.sizes,
      }}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Canvas>

## API

<ArgsTable of={DoubleMedia} />
