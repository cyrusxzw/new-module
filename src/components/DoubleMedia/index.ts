import { DoubleMedia } from './DoubleMedia';

export { DoubleMedia };
