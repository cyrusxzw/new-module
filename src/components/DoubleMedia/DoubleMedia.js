import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { MediaBlock } from './components/MediaBlock';
import styles from './DoubleMedia.module.css';

const DoubleMedia = ({ mediaOne, mediaTwo, className, theme }) => {
  const classSet = cx(styles.base, styles[theme], className);

  return (
    <div className={classSet}>
      {mediaOne && <MediaBlock {...mediaOne} theme={theme} />}
      {mediaTwo && <MediaBlock {...mediaTwo} theme={theme} />}
    </div>
  );
};

DoubleMedia.propTypes = {
  className: PropTypes.string,
  mediaOne: PropTypes.shape({
    altText: PropTypes.string,
    caption: PropTypes.node,
    fallbackImage: PropTypes.object,
    heading: PropTypes.string,
    link: PropTypes.shape({
      hasTargetInNewWindow: PropTypes.bool,
      text: PropTypes.string,
      type: PropTypes.string,
      url: PropTypes.string,
    }),
    poster: PropTypes.object,
    sizes: PropTypes.shape({
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
    }),
    type: PropTypes.oneOf(['video', 'image', 'scrollbasedvideo']),
  }),
  mediaTwo: PropTypes.shape({
    altText: PropTypes.string,
    caption: PropTypes.node,
    fallbackImage: PropTypes.object,
    heading: PropTypes.string,
    link: PropTypes.shape({
      hasTargetInNewWindow: PropTypes.bool,
      text: PropTypes.string,
      type: PropTypes.string,
      url: PropTypes.string,
    }),
    poster: PropTypes.object,
    sizes: PropTypes.shape({
      large: PropTypes.string,
      medium: PropTypes.string,
      small: PropTypes.string,
    }),
    type: PropTypes.oneOf(['video', 'image', 'scrollbasedvideo']),
  }),
  theme: PropTypes.oneOf(['dark', 'light']),
};

DoubleMedia.defaultProps = {
  className: undefined,
  mediaOne: undefined,
  mediaTwo: undefined,
  theme: 'dark',
};

export { DoubleMedia };
