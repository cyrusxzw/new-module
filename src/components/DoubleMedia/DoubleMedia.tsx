import React from 'react';
import cx from 'classnames';
import { MediaBlock } from './components/MediaBlock';
import type { DoubleMediaProps } from './DoubleMedia.types';
import styles from './DoubleMedia.module.css';

const DoubleMedia = ({
  className,
  mediaOne,
  mediaTwo,
  theme = 'dark',
}: DoubleMediaProps) => {
  const classSet = cx(styles.base, styles[theme], className);

  return (
    <div className={classSet}>
      {mediaOne && <MediaBlock {...mediaOne} theme={theme} />}
      {mediaTwo && <MediaBlock {...mediaTwo} theme={theme} />}
    </div>
  );
};

export { DoubleMedia };
