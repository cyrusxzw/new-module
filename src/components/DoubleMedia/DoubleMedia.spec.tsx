import React from 'react';
import { render } from '@testing-library/react';
import { DoubleMediaFixture } from './DoubleMedia.fixture';
import { DoubleMedia } from './DoubleMedia';
import { DoubleMediaProps } from './DoubleMedia.types';

describe('<DoubleMedia />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <DoubleMedia
        mediaOne={{
          heading: DoubleMediaFixture.title,
          type: DoubleMediaFixture.imageOne
            .type as DoubleMediaProps['mediaOne']['type'],
          altText: DoubleMediaFixture.imageOne.altText,
          caption: DoubleMediaFixture.imageOne.caption,
          fallbackImage: DoubleMediaFixture.video.fallbackImage,
          sizes: DoubleMediaFixture.imageOne.sizes,
        }}
        mediaTwo={{
          heading: DoubleMediaFixture.title,
          link: DoubleMediaFixture.link,
          type: DoubleMediaFixture.imageTwo
            .type as DoubleMediaProps['mediaTwo']['type'],
          altText: DoubleMediaFixture.imageTwo.altText,
          caption: DoubleMediaFixture.imageTwo.caption,
          fallbackImage: DoubleMediaFixture.video.fallbackImage,
          sizes: DoubleMediaFixture.imageTwo.sizes,
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
