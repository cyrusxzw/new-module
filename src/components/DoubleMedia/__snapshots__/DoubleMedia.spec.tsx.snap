// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DoubleMedia /> should render base component correctly 1`] = `
<div>
  <div
    class="base dark"
  >
    <figure
      class="base dark figure dark"
    >
      <picture
        class="base picture dark"
      >
        <source
          media="(min-width: 1025px)"
          srcset="./assets/images/DoubleImage/image-one-large.jpg"
        />
        <source
          media="(min-width: 640px)"
          srcset="./assets/images/DoubleImage/image-one-medium.jpg"
        />
        <source
          media="(min-width: 0px)"
          srcset="./assets/images/DoubleImage/image-one-small.jpg"
        />
        <img
          alt="A tactile approach infuses our work, from our product development to our retail service and visual merchandising."
          loading="lazy"
        />
      </picture>
      <figcaption
        class="figcaption"
      >
        <h3
          class="heading"
        >
          <span
            class="headingWrapper"
          >
            Aesop favourites
          </span>
        </h3>
        <div
          class="caption"
        >
          Our passion for creativity in form and function, extends to the setting of a meal among staff.
        </div>
      </figcaption>
    </figure>
    <figure
      class="base dark figure dark"
    >
      <a
        class="base left dark imageLinkWrapper"
        href="/au/c/skin/cleanse"
        target="_self"
      >
        <picture
          class="base picture dark"
        >
          <source
            media="(min-width: 1025px)"
            srcset="./assets/images/DoubleImage/image-two-large.jpg"
          />
          <source
            media="(min-width: 640px)"
            srcset="./assets/images/DoubleImage/image-two-medium.jpg"
          />
          <source
            media="(min-width: 0px)"
            srcset="./assets/images/DoubleImage/image-two-small.jpg"
          />
          <img
            alt="Image of: Our passion for creativity in form and function, extends to the setting of a meal among staff."
            loading="lazy"
          />
        </picture>
      </a>
      <figcaption
        class="figcaption"
      >
        <h3
          class="heading headingLink"
        >
          <a
            class="base left dark headingWrapper"
            href="/au/c/skin/cleanse"
            target="_self"
          >
            <span>
              Aesop favourites
            </span>
          </a>
        </h3>
        <div
          class="caption"
        >
          Our passion for creativity in form and function, extends to the setting of a meal among staff.
        </div>
      </figcaption>
    </figure>
  </div>
</div>
`;
